src/rttable.c

```diff
@@ -228,7 +240,7 @@ static void sendJoinLeaveUpstream(struct RouteTable* route, int join) {
 /**
 *   Clear all routes from routing table, and alerts Leaves upstream.
 */
-void clearAllRoutes(void) {
+void clearRoutes(struct IfDesc *IfDp) {
```

Comment:
unused function parameter?

Reply:
As of yet. The paramater will be a pointer to (old) interface configuration during dynamic interface evaluation to remove the routes belonging to (upstream) interface if it has been removed from configuration. I think it is better to use this function for such purpose as removeRoute() does all sorts of checks, while when interface is being removed from config you'll want to indiscriminately remove all routes going over that interface.

Reply:
Please introduce new variables only when you start really using it. Otherwise it is "dead code" (=unused). Also compiler would complain about declaring unused variable.

Reply:
fixed this, removed the parameter from funtion